src/runtime/statistics.py

```diff
@@ -138,6 +137,66 @@ def harmonic_mean(x: list[i32]) -> f64:
 
     return k / sum
 
+@overload
+def mode(x: list[i32]) -> i32:
+    """
+    Returns the mode of a data sequence of numbers
+    """
+    k: i32 = len(x)
+    if k == 0:
+        raise Exception("No mode for empty data")
+    i: i32
+    j: i32
+    ans: i32 = x[0]
+    for i in range(k):
+        for j in range(i + 1, k):
+            if x[i] > x[j]:
+                x[i], x[j] = x[j], x[i]
+
+    init: i32 = x[0]
+    count: i32 = 0
+    max_count: i32 = 0
+    c: i32
+    for c in range(k):
+        if x[c] == init:
+            count = count + 1
+            if max_count > count:
+                ans = x[c]
+        else:
+            init = x[c]
```

Comment:
The logic of this algorithm is wrong. This is an incorrect way of computing mode.